src/main/java/uk/gov/hmcts/cmc/claimstore/events/DocumentUploadEvent.java

```diff
@@ -0,0 +1,23 @@
+package uk.gov.hmcts.cmc.claimstore.events;
+
+import lombok.Value;
+import uk.gov.hmcts.cmc.claimstore.documents.output.PDF;
+import uk.gov.hmcts.cmc.domain.models.Claim;
+
+import java.util.List;
+
+import static com.google.common.collect.Lists.newArrayList;
+
+@Value
+public class DocumentUploadEvent {
```

Comment:
This class looks a ditto copy of `DocumentGeneratedEvent`, not sure we need a different class/event here. 

The events flow I can visualise for response document upload is:
`DefendantResponseEvent -> DocumentGeneratedEvent`
Then Handler/EventListener for `DocumentGeneratedEvent` will do the upload of the document.

Reply:
Yes,  DocumentUploadEvent contains the same information as DocumentGeneratedEvent , but DocumentGeneratedEvent is subscribed to by several listeners and I don't want those listeners to be invoked when the event is  to just upload a document.